src/tls/boring.rs

```diff
@@ -132,28 +142,27 @@ impl Certs {
         conn.check_private_key()?;
 
         // Ensure that client certificates are validated when present.
-        conn.set_verify(ssl::SslVerifyMode::FAIL_IF_NO_PEER_CERT);
-        conn.set_verify_callback(ssl::SslVerifyMode::PEER, |_, _| {
+        conn.set_verify_callback(Self::verify_mode(), Verifier::None.callback());
```

Comment:
is None TODO or expected long term?

Reply:
Long term for inbound I think. Unless we decide to add some extra validation. 

It also seems like it may be better to capture some verification errors outside of these callbacks; if we can pipe the san failures through boringssl we wouldn't need any callback here.